### ISSUE TITLE
[META] mvp for server-side views

### ISSUE THREAD

**Issue by hamilton** [iodide-project/iodide#669]
We want to support a very small set of views for our mvp server. They are:

- __user view__ - show a user's name, and all the notebooks that user has created. No sorting (yet), no filtering. Just a big `ul`.
- __home view__ - for now, let's just show all the notebooks available for now. Given that we're testing this ourselves, it'll be nice to see all the notebooks. So I see a grid or table w/ user name, date last edited, link to the notebook.
- __notebook view__ - this is the iodide SAP we've been building. Perhaps we have `meta+s` save this directly to the database now. Not sure how to approach this. the `?r=` is not necessary for an MVP - for now, a user only gets one notebook to save across.

These sets of views should live in the same project as @djbarnwal 's server work, and should be a separate set of code distinct from the iodide notebook itself. The notebook view, however, should serve the full notebook as-is, perhaps with some additional hooks for server features.

As for styling / design, I think it'd be fine if we skipped that for an MVP. If everything looks ugly but is functional, that's better than nothing, so I wouldn't sweat the css too much as a starting point.

Here is a whiteboard we drew up during the All-Hands w/ a rough sense of how this should all look.

![img_3354](https://user-images.githubusercontent.com/95735/41622492-9bcbc4fc-73c4-11e8-9580-e73d04988f32.JPG)

**Comment by djbarnwal** [comment]
@hamilton Sounds great! I'll direct my work towards this architecture.

**Comment by hamilton** [comment]
Great :) feel free to ping me or @wlach for any clarifications. I think we'll need the database side set up as per #633 before we can make these views / routes.

**Comment by djbarnwal** [comment]
@hamilton The database and API routes work is complete. I was wiring things with frontend at this point. Do you want me to push the server work or should I make a PR when everything gets completed?

**Comment by hamilton** [comment]
Seems like we should probably get some review on the db / routes, so perhaps a PR for that specifically would be worthwhile. @wlach will probably be taking on the review there, so I'll let him decide.

**Comment by wlach** [comment]
@djbarnwal you can always push your work-to-date right now, that way I can give some feedback today. it probably makes sense to test and then land things as a single unit though.

**Comment by djbarnwal** [comment]
@wlach I will push my work once PR #655 gets merged. My server component is built on top of the changes of that PR (which has been changed further). Rebasing now would mess things up. I'll rebase and push my work once that PR is merged to avoid any git jargon.

**Comment by wlach** [comment]
One thing we discussed between ourselves (me, @hamilton, @bcolloran) when I was in SF but didn't mention here is that we still want to support running iodide notebooks in “standalone” mode-- that is, self-contained pieces of html and jsmd that have no connection to a particular iodide server.

The way we envisioned this is that when building the server, we would produce two self-contained react/webpack bundles for the frontend pieces:

* One for the notebook itself
* One for everything else (initially viewing all notebooks and user notebooks, as described above)

We would then (somehow) have enough intelligence on the server end of things to serve the right kind of content to the user based on where they have currently navigated to: this will probably require some combination of hackery with the express webserver we're using and some kind of tool like [react-router](https://reacttraining.com/react-router/). 

For a very very first pass, it is probably ok to handle loading/rendering the notebook outside of the context of a user (e.g. `http://extremely-alpha.iodide.io/notebook?jsmd=http://extremely-alpha.iodide.io/api/...`) but longer term we would want the more friendly url like `http://extremely-alpha.iodide.io/user/<notebook id>`

**Comment by djbarnwal** [comment]
@wlach @hamilton @bcolloran Should we create a new repository to host the code for Iodide server and website. This would help us with a more organized file structure. This way we can keep the notebook component clean and away from all the clutter.

**Comment by bcolloran** [comment]
@djbarnwal -- when @wlach and @hamilton and i discussed this last week, we decided that in the short term it might make most sent to keep the server code and the notebook code in the same repo, because changes in the server might also require changes in the notebook UI, so for now it might be easier to manage our test deployments if these things are in sync within one repo. but we can certainly revisit that in a month or two, or we can change our minds now if you think its a bad idea.

in terms of managing the clutter: perhaps we can just start with moving all the server code into a separate top-level `iodide-server` directory? i think that should be enough to keep things clean for now.

**Comment by bcolloran** [comment]
closing -- this issue has been superseded by more current issues
